docs/howtos/DataPlane Generation - DPG/09tsp_client.mdx

```diff
@@ -0,0 +1,182 @@
+# Getting started with `tsp-client`
+
+`tsp-client` is a simple command line tool to facilitate generating client libraries from TypeSpec.
+
+## Installation
+
+```
+npm install -g @azure-tools/typespec-client-generator-cli
```

Comment:
Should that tool just be added to the spec repo `package.json` if the main goal is to use it there?

Reply:
I dont think we require tsp-client for any part of the REST API spec development process currently. We only use it to do initial conversion for some brownfield services. Aside from that it is mainly used in the language repositories and their corresponding automation scripts (this last part is in progress).

Reply:
yeah but isn't the folow that you'd want ot call `tsp-client` from the spec repo when you are ready to generate sdks?

Reply:
Through the release planner we link folks to this page when they're ready to generate sdks: https://eng.ms/docs/products/azure-developer-experience/develop/sdk-develop?tabs=dataplane That page links folks to language specific generation instructions, which work within the language repos. Plus some languages, like C# and Python wrap tsp-client under language specific tooling. I believe C# requires their sdks to be generated with the dotnet build command since it does some other processing under the hood for them so that the generated sdks are compatible with the configurations used in the repo

Reply:
We might still need to add tsp-client in the package.json once we integrate it more with the specs repo, since we want to have the automation scripts use tsp-client. Once we do that I can revisit and update this page

Reply:
I could also remove this from the page and recommend folks run it with npx

Reply:
no that's fine if the goal is do have it there in the future but you are not ready yet docs looks good then